From 814f6ef31ce83f9ba0c60eac825bd7d94dd2ced6 Mon Sep 17 00:00:00 2001
From: valtzu <valtzu@gmail.com>
Date: Mon, 9 Jan 2023 04:56:40 +0200
Subject: [PATCH] Platform/RPi/RPiFirmwareDxe: Fix RPi4 serial number

Fetching serial number didn't work properly due to compiler optimization / struct padding.

Signed-off-by: valtzu <valtzu@gmail.com>
---
 Platform/RaspberryPi/Drivers/RpiFirmwareDxe/RpiFirmwareDxe.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Platform/RaspberryPi/Drivers/RpiFirmwareDxe/RpiFirmwareDxe.c b/Platform/RaspberryPi/Drivers/RpiFirmwareDxe/RpiFirmwareDxe.c
index 4edec0ad..984083f7 100644
--- a/Platform/RaspberryPi/Drivers/RpiFirmwareDxe/RpiFirmwareDxe.c
+++ b/Platform/RaspberryPi/Drivers/RpiFirmwareDxe/RpiFirmwareDxe.c
@@ -340,7 +340,7 @@ RpiFirmwareGetMacAddress (
   return EFI_SUCCESS;
 }
 
-#pragma pack()
+#pragma pack(1)
 typedef struct {
   UINT64                    Serial;
 } RPI_FW_SERIAL_TAG;
-- 
2.34.1

